src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -80,10 +80,10 @@ public void EndEvent(object eventHandle)
             SystemMetricEvent evt = eventHandle as SystemMetricEvent;
             if (evt != null)
             {
-                long latencyMS = 0;
-                evt.StopWatch.Stop();
+                long latencyMS;
                 if (evt.StopWatch != null)
                 {
+                    evt.StopWatch.Stop();
```

Comment:
Because StopWatch is never null this won't have any real effect because we always set it when creating SystemMetricEvents above - you're just improving the logic because we're already checking for null?

Reply:
This was my point. There's a check, let it check what it checks 😅